README.md

```diff
@@ -25,10 +25,79 @@ For more information on each of these steps, follow the link to their respective
+
+To instantiate a scenario, simply provide a set of variables and a(n optional) set of constraints, e.g.
+```
+from causal_testing.specification.variable import Input, Output, Meta
+from causal_testing.specification.scenario import Scenario
+
+x = Input("x", int)  # Define an input with name "x" of type int
+y = Output("y", float)  # Define an output with name "y" of type float
+z = Output("y", int)  # Define a meta with name "z" of type int
+
+modelling_scenario = Scenario({x, y, z}, {x > z, z < 3})  # Define a scenario with the three variables and two constraints
+```
+
+Note that scenario constraints are primarily intended to help specify the region of the input space under test in a manner consistent with the Category Partition Method. It is not intended to serve as a test oracle. Use constraints sparingly and with caution to avoid introducing data selection bias. We use Z3 to handle constraints. For help with this, check out [their documentation](https://ericpony.github.io/z3py-tutorial/guide-examples.htm).
+
+Having fully specified the modelling scenario, we are now ready to test. Causal tests are, essentially [metamorphic tests](https://en.wikipedia.org/wiki/Metamorphic_testing) which are executed using statistical causal inference. A causal test expresses the change in a given output that we expect to see when we change a particular input in a particular way, e.g.
+```
+from causal_testing.testing.causal_test_case import CausalTestCase
+from causal_testing.testing.causal_test_outcome import Positive
+
+causal_test_case = CausalTestCase(
+    control_input_configuration={x: 0},  # Our unmodified (or 'control') configuration in which input x is 0
+    treatment_input_configuration={x: 1}, # Our modified (or 'treatment') configuration in which input x is 1
+    expected_causal_effect=Positive,  # We expect to see a positive change as a result of this
+    outcome_variables=[y],  # We expect to observe that positive change in variable y
+)
+```
+
+Before we can run our test case, we first need data. There are two ways to acquire this: 1. run the model with the specific input configurations we're interested in, 2. use data from previous model runs. For a small number of specific tests where accuracy is critical, the first approach will yield the best results. To do this, you need to instantiate the `ExperimentalDataCollector` class. Further details on this coming soon.
+
+Where there are many test cases use pre-existing data is likely to be faster. If the program's behaviour can be estimated statistically, the results should still be reliable as long as there is enough data for the estimator to work as intended. This will vary depending on the program and the estimator. To use this method, simply instantiate the `ObservationalDataCollector` class with the modelling scenario and a path to the CSV file containing the runtime data, e.g.
```

Comment:
The first sentence might read nicer as "Where there are many test cases, using pre-existing data is likely..."?